harvester_e2e_tests/integration/test_images.py

```diff
@@ -0,0 +1,375 @@
+    assert 200 == code, (code, data)
+    assert unique_name == data['metadata']['name']
+
+
+def create_with_invalid_url(api_client, unique_name):
+    code, data = api_client.images.create_by_url(unique_name, f"https://{unique_name}.img")
+
+    assert 201 == code, (code, data)
+
+    endtime = datetime.now() + timedelta(minutes=3)
+    while endtime > datetime.now():
+        code, data = api_client.images.get(unique_name)
+        image_conds = data.get('status', {}).get('conditions', [])
+        if len(image_conds) > 0:
+            break
+        sleep(3)
+
+    assert len(image_conds) == 1, f"Got unexpected image conditions!\n{data}"
+    assert "Initialized" == image_conds[0].get("type")
+    assert "False" == image_conds[0].get("status")
+    assert "no such host" in image_conds[0].get("message")
+
+    api_client.images.delete(unique_name)
+
+
+class TestBackendImages:
+
+    @pytest.mark.images_p1
+    @pytest.mark.dependency(name="create_image_from_volume")
+    def test_create_image_from_volume(self, api_client, request,
+                                      unique_name, export_storage_class):
+        """
+        Test create image from volume
+
+        Steps:
+        1. Create a volume "test-volume" in Volumes page
+        2. Export the volume to image "export-image"
+        3. Check the image "export-image" exists
+        4. Cleanup image "export-image" on Images page
+        5. Cleanup volume "test-volume" on Volumes page
+        """
+        wait_timeout = request.config.getoption('--wait-timeout')
+
+        spec = api_client.volumes.Spec(1)
+        code, data = api_client.volumes.create(unique_name, spec)
+
+        assert 201 == code, (code, data)
+
+        get_volume(api_client, unique_name)
+
+        api_client.volumes.export(unique_name, unique_name, export_storage_class)
+
+        image_id = get_export(api_client, unique_name, wait_timeout)
+
+        delete_image(api_client, image_id, wait_timeout)
+
+    @pytest.mark.images_p1
+    @pytest.mark.dependency(name="create_image_url")
+    def test_create_image_url(self, api_client, request, iso_image_url,
+                              qcow2_image_url, qcow2_name, iso_name):
+        """
+        Test create raw and iso type image from url
+
+        Steps:
+        1. Open image page and select default URL
+        2. Input qcow2 image file download URL, wait for download complete
+        3. Check the qcow2 image exists
+        4. Input iso image file download URL, wait for download complete
+        5. Check the iso image exists
+        """
+        wait_timeout = request.config.getoption('--wait-timeout')
+
+        create_image_url(api_client, qcow2_name, qcow2_image_url, wait_timeout)
+        create_image_url(api_client, iso_name, iso_image_url, wait_timeout)
+
+    @pytest.mark.images_p1
+    @pytest.mark.dependency(name="delete_image_recreate", depends=["create_image_url"])
+    def test_delete_image_recreate(self, api_client, request,
```

Comment:
This test case should also use pytest.mark.parametrize or parameterized fixture.